common/persistence/health_signal_aggregator.go

```diff
@@ -44,7 +44,7 @@ const (
 type (
 	HealthSignalAggregator interface {
 		common.Daemon
-		Record(callerSegment int32, latency time.Duration, err error)
+		Record(callerSegment int32, namespace string, latency time.Duration, err error)
```

Comment:
nit: may need to define a struct for all the parameters here.

Reply:
I want to avoid a situation like we have with CallerInfo where we have a struct that is passed around many places but only some fields are used in each. If we need to add another parameter in the future, then yeah I think it makes sense to define a struct or reuse CallerInfo, but for now I don't think 4 is too bad.